xtask/src/commands/release.rs

```diff
@@ -127,6 +123,49 @@ impl Prepare {
         Ok(())
     }
 
+    fn ensure_pristine_checkout(&self) -> Result<(), anyhow::Error> {
+        let git = which::which("git")?;
+        let output = std::process::Command::new(git)
+            .args(["status", "--untracked-files=no", "--porcelain"])
+            .output()?;
+
+        if !output.stdout.is_empty() {
+            return Err(anyhow!(
+                "git workspace was not clean and requires 'git stash' before releasing"
+            ));
+        }
+        Ok(())
+    }
+
+    fn ensure_prereqs(&self) -> Result<()> {
+        if which::which("git").is_err() {
+            return Err(anyhow!(
+                "the 'git' executable could not be found in your PATH"
+            ));
+        }
+
+        if which::which("helm").is_err() {
+            return Err(anyhow!("the 'helm' executable could not be found in your PATH.  Install it using the instructions at https://helm.sh/docs/intro/install/ and try again."));
+        }
+
+        if which::which("helm-docs").is_err() {
+            return Err(anyhow!("the 'helm-docs' executable could not be found in your PATH.  Install it using the instructions at https://github.com/norwoodj/helm-docs#installation and try again."));
+        }
+
+        if which::which("cargo-about").is_err() {
+            return Err(anyhow!("the 'cargo-about' executable could not be found in your PATH.  Install it by running `cargo install --locked cargo-about"));
+        }
+
+        if which::which("cargo-deny").is_err() {
+            return Err(anyhow!("the 'cargo-deny' executable could not be found in your PATH.  Install it by running `cargo install --locked cargo-deny"));
+        }
+
+        // if std::env::var("GITHUB_TOKEN").is_err() {
```

Comment:
To delete ?

Reply:
Ah, actually, I think this could also be enforced ahead of time.

Reply:
To uncomment, perhaps!  I'm actually not convinced that we want `GITHUB_TOKEN` to need to be set, rather than just using local tooling that already has such a thing set (like[ the `gh` CLI](https://cli.github.com/)) and stored, but we can discuss that outside the scope of this PR.